python-package/lightgbm/callback.py

```diff
@@ -209,12 +215,69 @@ def _init(env):
                 best_score.append(float('inf'))
                 cmp_op.append(lt)
 
+    def _metric_alias_matching(target_metric):
+        """Convert metric name for early stopping from alias to representative in order to match the param metric."""
+        metrics_list = [("l1", {"mean_absolute_error", "mae", "regression_l1"}),
+                        ("l2", {"mean_squared_error", "mse", "regression_l2", "regression"}),
+                        ("l2_root", {"root_mean_squared_error", "rmse"}),
+                        ("mape", {"mean_absolute_percentage_error"}),
+                        ("ndcg", {"lambdarank"}),
+                        ("map", {"mean_average_precision"}),
+                        ("binary_logloss", {"binary"}),
+                        ("multi_logloss",
+                        {"multiclass", "softmax", "multiclassova", "multiclass_ova", "ova", "ovr"}),
+                        ("xentropy", {"cross_entropy"}),
+                        ("xentlambda", {"cross_entropy_lambda"}),
+                        ("kldiv", {"kullback_leibler"})]
+        for metric, aliases in metrics_list:
+            if target_metric in aliases:
+                return metric
+        return target_metric
+
     def _callback(env):
         if not cmp_op:
             _init(env)
         if not enabled[0]:
             return
+        if first_metric_only:
```

Comment:
@matsuken92 Am I right about that this code is executed at each boosting iteration? If so, than it seems to be quite inefficient. I guess that determining of the needed `eval_metric` can be performed only once. Is it possible to move this part of code to `_init` phase or (if not) execute only once?